src/main/java/meditracker/medication/Medication.java

```diff
@@ -200,14 +316,20 @@ public void revertMedication(Medication medication) {
     public static Medication deepCopy(Medication medication) {
         Medication newMedication = new Medication();
 
-        newMedication.setName(medication.getName());
+        try {
+            newMedication.setName(medication.getName());
+            newMedication.setRepeat(medication.getRepeat());
+        } catch (MediTrackerException e) {
+            // critical error as this should not happen
+            throw new RuntimeException(e);
```

Comment:
Should we try to handle this gracefully instead of crashing?

Reply:
just like how the remarks have a default "nil" value if null

Reply:
Resolved in eb0bf3c